demos/complementary-color-prediction/complementary-color-prediction.ts

```diff
@@ -155,36 +155,39 @@ class ComplementaryColorModel {
    * to pass data into the model. Generates `exampleCount` data points.
    */
   private generateTrainingData(exampleCount: number) {
-    this.math.scope(() => {
-      const rawInputs = new Array(exampleCount);
-      for (let i = 0; i < exampleCount; i++) {
-        rawInputs[i] = [
-          this.generateRandomChannelValue(), this.generateRandomChannelValue(),
-          this.generateRandomChannelValue()
-        ];
-      }
-
-      // Store the data within Array1Ds so that learnjs can use it.
-      const inputArray: Array1D[] =
-          rawInputs.map(c => Array1D.new(this.normalizeColor(c)));
-      const targetArray: Array1D[] = rawInputs.map(
-          c => Array1D.new(
-              this.normalizeColor(this.computeComplementaryColor(c))));
-
-      // This provider will shuffle the training data (and will do so in a way
-      // that does not separate the input-target relationship).
-      const shuffledInputProviderBuilder =
-          new InCPUMemoryShuffledInputProviderBuilder(
-              [inputArray, targetArray]);
-      const [inputProvider, targetProvider] =
-          shuffledInputProviderBuilder.getInputProviders();
-
-      // Maps tensors to InputProviders.
-      this.feedEntries = [
-        {tensor: this.inputTensor, data: inputProvider},
-        {tensor: this.targetTensor, data: targetProvider}
+    const rawInputs = new Array(exampleCount);
+    const oldMath = ENV.math;
+    const safeMode = false;
+    const math = new NDArrayMath('cpu', safeMode);
+    ENV.setMath(math);
```

Comment:
Why are we using for the following computations the CPU, but set the math context in the end of this class method to the old math reference (GPU?) again? Looking at it without being into it, it seems like a temporary workaround?